catroid/src/org/catrobat/catroid/ui/dialogs/NewDataDialog.java

```diff
@@ -89,6 +90,17 @@ public NewDataDialog(Spinner spinnerToUpdate, DialogType dialogType) {
 	public void onCancel(DialogInterface dialog) {
 		super.onCancel(dialog);
 		userListDialogListenerListFinishNewUserListDialog(null);
+		if (userVariableIfCancell != null) {
+			int spinnerSize = spinnerToUpdate.getCount() - 1;
+			for (int spinnerIndex = 1; spinnerIndex <= spinnerSize; spinnerIndex++) {
+				spinnerToUpdate.setSelection(spinnerIndex);
+				if (spinnerToUpdate.getSelectedItem().equals(userVariableIfCancell)) {
```

Comment:
UserVariableAdapterWrapper offers method getPositionOfItem() that does exactly this on a spinner. also if spinner id is passed instead of the userVariable this could also be skipped.